lib/rucio/core/did.py

```diff
@@ -488,6 +488,18 @@ def attach_dids_to_dids(attachments, account, ignore_duplicate=False, session=No
     parent_did_condition = list()
     parent_dids = list()
     for attachment in attachments:
+        content = [{'scope': did['scope'], 'name': did['name']} for did in list_content(scope=attachment['scope'], name=attachment['name'], session=session)]
```

Comment:
This is getting too expensive on the server side, because it essentially adds O(n) queries for n dids to be attached, this is really a big increase for attaches. I think it is better to leave the server-side code simple and just let it try to insert and run into an IntegrityException
This entire block (491-502) should be removed. 

We should do this on bin/rucio side only.
Thus when you have to do the chunking: If the IntegrityException is raised, then you check for this chunk which files have already been added to the dataset and which don't, then you go to the next chunk and try to insert it etc.
For now I would even say to only do this this for the chunking and leave the non-chunked mode (eg. you only attach 5 dids) as it is already, just let it fail without trying to complete it.